Clear intermediate draws

pedrobmarin opened the pull request:
Trying to fix #26 .

Could you test these changes @hiroshisuga ?

hiroshisuga commented:
Hi @pedrobmarin ,
Not fixed.
Three screen captures represent my recording at 50:12 on (1) your fixed version, (2) playback2.0, and (3) my current fixed version. 
![1](https://user-images.githubusercontent.com/45039819/85089506-c78e2a00-b21d-11ea-81a2-8b7d9add78df.jpg)
![2](https://user-images.githubusercontent.com/45039819/85089510-ca891a80-b21d-11ea-9e6e-546f62513723.jpg)
![3](https://user-images.githubusercontent.com/45039819/85089516-cc52de00-b21d-11ea-8b9f-4fea8ea08057.jpg)

pedrobmarin commented:
I refactored your changes to replace the whole 2 indexes control because the multi-user whiteboard makes everything more complicated, indeed.
 - included a control to avoid checking the entire array based on the current time. I don't think we could improve much more than that since the data can have clear holes at any time.
 - monitored the CPU usage after this changes and looks like this does not add too much more load to the process
 - changed `getCurrentDataInterval` tests.

I'll be closing your PR if you don't mind, @hiroshisuga .

hiroshisuga commented:
Thanks, and I don't mind closing the PR.
